scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/types/ConverterProvider.scala

```diff
@@ -180,17 +180,18 @@ private[types] object ConverterProvider {
         case t if t =:= typeOf[String]              => tree
 
         case t if t =:= typeOf[BigDecimal] =>
-          q"_root_.com.spotify.scio.bigquery.Numeric($tree).toString"
+          q"_root_.com.spotify.scio.bigquery.Numeric.bytes($tree)"
         case t if t =:= typeOf[ByteString] =>
           q"_root_.java.nio.ByteBuffer.wrap($tree.toByteArray)"
         case t if t =:= typeOf[Array[Byte]] =>
           q"_root_.java.nio.ByteBuffer.wrap($tree)"
 
-        case t if t =:= typeOf[Instant] => q"$tree.getMillis * 1000"
+        case t if t =:= typeOf[Instant] =>
+          q"_root_.com.spotify.scio.bigquery.Timestamp.micros($tree)"
         case t if t =:= typeOf[LocalDate] =>
-          q"_root_.com.spotify.scio.bigquery.Date($tree)"
+          q"_root_.com.spotify.scio.bigquery.Date.days($tree)"
         case t if t =:= typeOf[LocalTime] =>
-          q"_root_.com.spotify.scio.bigquery.Time($tree)"
+          q"_root_.com.spotify.scio.bigquery.Time.micros($tree)"
```

Comment:
so it was returning String before?

Reply:
Indeed, by changing [this](https://github.com/spotify/scio/pull/5529/files#diff-39ac37a4ec2c09860762dae1c3467ca60ce4418840615f5fc0dac3426bffbd45R442), I created a behavior change.

When extracting from avro, we disabled the logical types -> those types are returned as STRING (mentioned [here](https://github.com/spotify/scio/blob/cc1e584206e70b3e88dffcad9ea7b6f4a0d7e8b7/scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/syntax/ScioContextSyntax.scala#L81)). This is very annoying because the same record can't be written back in the table.

Setting `useAvroLogicalTypes` on the IO enables that. This will however be breaking for all pipelines reading BQ `DATE`, `TIME` and `DATETIME` that expects `string` and will get logical type.

This is also not in sync with `storage` API that uses logical types for such columns.

Reply:
Updated the PR to keep the same behavior.
Introduced new `Format.GenericRecordWithLogicalType` to enable BQ avro read with desired setup.
Change `BigQueryTyped.Table` to always use logical types